specfile/tags.py

```diff
@@ -0,0 +1,439 @@
+    "Transfiletriggerscripts",
+    "Transfiletriggertype",
+    "Transfiletriggerun",
+    "Transfiletriggerversion",
+    "Triggerconds",
+    "Triggerflags",
+    "Triggerin",
+    "Triggerindex",
+    "Triggername",
+    "Triggerpostun",
+    "Triggerprein",
+    "Triggerscriptflags",
+    "Triggerscriptprog",
+    "Triggerscripts",
+    "Triggertype",
+    "Triggerun",
+    "Triggerversion",
+    "Url",
+    "V",
+    "Vcs",
+    "Vendor",
+    "Verbose",
+    "Verifyscript",
+    "Verifyscriptflags",
+    "Verifyscriptprog",
+    "Veritysignaturealgo",
+    "Veritysignatures",
+    "Version",
+    "Xpm",
+]
+
+
+class Tag:
+    """
+    Class that represents spec file tag.
+
+    Attributes:
+        name: Name of the tag.
+        value: Literal value of the tag as stored in the spec file.
+        comments: List of comment lines associated with the tag.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        value: str,
+        expanded_value: str,
+        separator: str,
+        preceding_lines: List[str],
+    ) -> None:
+        if not name or name.capitalize().rstrip("0123456789") not in TAG_NAMES:
+            raise ValueError(f"Invalid tag name: '{name}'")
+        self.name = name
+        self.value = value
+        self._expanded_value = expanded_value
+        self._separator = separator
+        self.comments, self._preceding_lines = self._extract_comments(preceding_lines)
+
+    def __repr__(self) -> str:
+        preceding_lines = repr(self._preceding_lines)
+        return (
+            f"Tag('{self.name}', '{self.value}', '{self._expanded_value}', "
+            f"'{self._separator}', {preceding_lines})"
+        )
+
+    @property
+    def valid(self) -> bool:
+        """Validity of the tag. A tag is valid if it 'survives' the expansion of the spec file."""
+        return self._expanded_value is not None
+
+    @property
+    def expanded_value(self) -> str:
+        """Value of the tag after expanding macros and evaluating all conditions."""
+        return self._expanded_value
+
+    @staticmethod
+    def _extract_comments(lines: List[str]) -> Tuple[List[str], List[str]]:
+        comments = itertools.takewhile(lambda x: x.startswith("#"), reversed(lines))
+        rest = itertools.dropwhile(lambda x: x.startswith("#"), reversed(lines))
+        return [c[1:] for c in reversed(list(comments))], list(reversed(list(rest)))
+
+
+class Tags(collections.UserList):
+    """
+    Class that represents all tags in a certain %package section.
+
+    Tags can be accessed by index or conveniently by name as attributes:
+    ```
+    # print name of the first tag
+    print(tags[0].name)
+
+    # set value of Url tag
+    tags.url = 'https://example.com'
+
+    # remove Source1 tag
+    del tags.source1
+    ```
+
+    Attributes:
+        data: List of individual tags.
+    """
+
+    def __init__(
+        self, data: Optional[List[Tag]] = None, remainder: Optional[List[str]] = None
```

Comment:
Maybe some documentation about what `remainder` is would be nice.

Reply:
It's an internal attribute, and nobody should be instantiating `Tags` directly, but I'll add some description for it.